### ISSUE TITLE
pecl module defaults is broken

### ISSUE THREAD

**Issue by ryan-lane** [saltstack/salt#24553]
```
[INFO    ] Executing command "yes $'\\n' | pecl -d preferred_state=stable install mongo" in directory '/root'
```

```
# yes $'\\n' | pecl -d preferred_state=stable install mongo
downloading mongo-1.6.9.tgz ...
Starting to download mongo-1.6.9.tgz (208,955 bytes)
.............................................done: 208,955 bytes
[snip]

configure: error: sasl.h not found!
ERROR: `/tmp/pear/temp/mongo/configure --with-mongo-sasl=\n' failed
```

Looks like \n is a bad idea. This works, though:

```
# yes '' | pecl -d preferred_state=stable install mongo
[snip]
Build process completed successfully
Installing '/usr/lib/php5/20121212/mongo.so'
install ok: channel://pecl.php.net/mongo-1.6.9
configuration option "php_ini" is not set to php.ini location
You should add "extension=mongo.so" to php.ini
```

**Comment by ryan-lane** [comment]
The worst part about the \n failure above is that it's completely silent, because pecl returns a 0 exit code.

**Comment by ryan-lane** [comment]
Fixed in #24554

**Comment by rallytime** [comment]
Thanks for the report and for the fix, as usual, @ryan-lane!
